torch/csrc/jit/import_source.cpp

```diff
@@ -218,7 +201,15 @@ struct SourceImporterImpl : public Resolver,
     if (it != env_.end()) {
       return it->second;
     }
-
+    auto graph = m.graph();
```

Comment:
Why do we need this? It should just call `aten::Float` which already has this implemented

Reply:
`aten::Float` doesn't handle `nan`. since this is resolving the `inf` as a name it's not bound in as a value yet. to use `float` we would have to insert a str value then insert `aten::Float` on it. not sure that's better.

Reply:
`aten::Float` should be handling NaN since it’s valid Python. `inf` shouldn’t be a valid name, if the exporter is using it as one we should fix it there to be `float(“inf”)`

Reply:
If i make it not a valid name, then it breaks BC. I don't think that's worth it.

Reply:
You can insert it as a string value and add the Float call to keep things working

Reply:
Then you have to separate pathways for re-importing: one in which we have inf as float("inf") and another where we just have "inf". Is that worth the additional complexity compared to the present solution ?

Reply:
We already have two separate pathways of interpreting “inf”, one of which is using it as a value which is wrong since it’s not valid Python

Reply:
`inf` is valid python:
`from torch._six import inf; inf`

Reply:
That's just an imported name (which we don't have in the exported code afaik), it's defined like this in `_six.py`

```python
if PY2:
    inf = float('inf')
    nan = float('nan')
else:
    import math
    inf = math.inf
    nan = math.nan
```

Reply:
AFAIK "inf" or "nan" is not something we generate anymore, i can't find it anywhere in python_print. so there's no point in "fixing" the export, it's already fixed. we just need to support the existing behavior.

Reply:
I don't know if we ever did... it definitely seems like a weird thing to do and with v1.0 and v1.1 for
```python
@torch.jit.script
def fn(x):
    return float('inf')

fn.save('myfn.zip')
```

we generate
```python
op_version_set = 0
def forward(self,
    x: Tensor) -> float:
  return float("inf")
```
so there might not even be a backwards compatibility problem.

So if that's the case then there's no need for special handling of `inf` or `nan` anywhere outside of the `aten::Float` op

Reply:
Okay that's 1.0 and 1.1, but it doesn't mean we never did. I can send you the test internally that is failing because we reassign "inf" with a variable. If you want to debug it you can. I'm not sure it's worth your time.